test/hpr_http_roaming_packet_SUITE.erl

```diff
@@ -0,0 +1,1588 @@
+            <<"PHYPayload">> => hpr_http_roaming_utils:binary_to_hexstring(
+                hpr_packet_up:payload(PacketUp1)
+            ),
+            <<"ULMetaData">> => #{
+                <<"DevAddr">> => ?DEVADDR_ACTILITY_BIN,
+                <<"DataRate">> => hpr_lorawan:datarate_to_index(
+                    Region,
+                    hpr_packet_up:datarate(PacketUp1)
+                ),
+                <<"ULFreq">> => hpr_packet_up:frequency_mhz(PacketUp1),
+                <<"RFRegion">> => erlang:atom_to_binary(Region),
+                <<"RecvTime">> => hpr_http_roaming_utils:format_time(GatewayTime1),
+
+                <<"FNSULToken">> => fun erlang:is_binary/1,
+                <<"GWCnt">> => 2,
+                <<"GWInfo">> => [
+                    #{
+                        <<"ID">> => hpr_http_roaming_utils:binary_to_hexstring(
+                            hpr_utils:pubkeybin_to_mac(PubKeyBin1)
+                        ),
+                        <<"RFRegion">> => erlang:atom_to_binary(Region),
+                        <<"RSSI">> => hpr_packet_up:rssi(PacketUp1),
+                        <<"SNR">> => hpr_packet_up:snr(PacketUp1),
+                        <<"DLAllowed">> => true
+                    },
+                    #{
+                        <<"ID">> => hpr_http_roaming_utils:binary_to_hexstring(
+                            hpr_utils:pubkeybin_to_mac(PubKeyBin2)
+                        ),
+                        <<"RFRegion">> => erlang:atom_to_binary(Region),
+                        <<"RSSI">> => hpr_packet_up:rssi(PacketUp2),
+                        <<"SNR">> => hpr_packet_up:snr(PacketUp2),
+                        <<"DLAllowed">> => true
+                    }
+                ]
+            }
+        }),
+
+    %% Gateway with better RSSI should be chosen
+    ?assertMatch(
+        {ok, TransactionID, 'US915', PacketTime1, <<"127.0.0.1:3002/uplink">>, sync},
+        hpr_http_roaming:parse_uplink_token(Token)
+    ),
+
+    ok.
+
+http_multiple_joins_same_dest_test(_Config) ->
+    DevEUI1 = 1,
+    AppEUI1 = 16#200000001,
+
+    #{secret := PrivKey, public := PubKey} = libp2p_crypto:generate_keys(ecc_compact),
+    SigFun = libp2p_crypto:mk_sig_fun(PrivKey),
+    PubKeyBin = libp2p_crypto:pubkey_to_bin(PubKey),
+
+    {ok, _Pid} = hpr_http_roaming_sup:start_link(),
+
+    GatewayTime = erlang:system_time(millisecond),
+    PacketUp = test_utils:frame_packet_join(
+        PubKeyBin,
+        SigFun,
+        DevEUI1,
+        AppEUI1,
+        #{timestamp => GatewayTime}
+    ),
+
+    join_test_route(DevEUI1, AppEUI1, sync, ?NET_ID_ACTILITY),
+    join_test_route(DevEUI1, AppEUI1, sync, ?NET_ID_ORANGE),
+
+    ok = start_uplink_listener(#{port => 3002, callback_args => #{forward => self()}}),
+
+    ok = hpr_routing:handle_packet(PacketUp),
+
+    {ok, #{<<"ReceiverID">> := ReceiverOne}, _, _} = http_rcv(),
+
+    lager:debug("ReceiverOne: ~p", [ReceiverOne]),
```

Comment:
We don't generally `lager` log in tests. 
@macpie is there a reason you don't, or just happenstance?

Reply:
Leftover debugging statements. I will remove them.